client/src/App.js

```diff
@@ -29,55 +30,60 @@ function App() {
     fetchProducts();
   }, []);
 
+  useEffect(() => {
+    const saveCartToLocalStorage = () => {
```

Comment:
Why are you saving items into the cart on component mount?

Reply:
I put cart as a dependency, so i think it happens only when cart changes.

Reply:
okay, I missed that